src/overlays/actors/ovl_Oceff_Storm/z_oceff_storm.c

```diff
@@ -33,22 +31,183 @@ const ActorInit Oceff_Storm_InitVars = {
     (ActorFunc)OceffStorm_Draw,
 };
 
-#endif
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_8098176C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981928.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981B48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/func_80981BB8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Oceff_Storm/OceffStorm_Draw.s")
+void OceffStorm_SetupAction(OceffStorm* this, OceffStormActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+s32 func_8098176C(PlayState* play) {
+    s32 phi_v1 = false;
+
+    switch (play->sceneNum) {
+        case SCENE_13HUBUKINOMITI:
+        case SCENE_11GORONNOSATO:
+        case SCENE_10YUKIYAMANOMURA:
+        case SCENE_14YUKIDAMANOMITI:
+        case SCENE_12HAKUGINMAE:
+        case SCENE_17SETUGEN:
+        case SCENE_GORONRACE:
+            if (gSaveContext.sceneSetupIndex == 0) {
+                phi_v1 = true;
+            }
+            break;
+        case SCENE_10YUKIYAMANOMURA2:
+            if (gSaveContext.sceneSetupIndex == 1) {
+                phi_v1 = true;
+            }
+            break;
+    }
+    return phi_v1;
```

Comment:
`ret` will do. Looks like this determines whether to not spawn the extra effect actor?